pinggrapher.go

```diff
@@ -100,25 +99,21 @@ func NewStats(times []float64, timestamp int64) Stats {
 	}
 }
 
-func read(pings chan float64) {
+func read(pings chan float64, host string) {
 	go func() {
-		var line string
-		var err error
-		reader := bufio.NewReader(os.Stdin)
-		for {
-			line, err = reader.ReadString('\n')
-			if err == io.EOF {
-				return
-			}
-			if err != nil {
-				log.Fatal(err)
-			}
-			ping, err := strconv.ParseFloat(strings.TrimSpace(line), 64)
-			if err != nil {
-				log.Printf("Couldn't convert '%s': %s", line, err)
-			}
+		log.Printf("Pinging %v", host)
+		pinger, err := ping.NewPinger(host)
+		if err != nil {
+			panic(err)
+		}
+		pinger.Count = 0
+		pinger.Interval = time.Second * 1
+		pinger.Timeout = time.Second * 5
+		pinger.OnRecv = func(pkt *ping.Packet) {
+			ping := float64(pkt.Rtt / time.Millisecond)
 			pings <- ping
 		}
+		pinger.Run() // blocks until finished
```

Comment:
`pinggrapher google.com` is more convenient than `ping 192.168.0.1 | awk  '/from/ { split($7, resArr, "="); print resArr[2] }' | pinggrapher`